stl/inc/algorithm

```diff
@@ -3520,11 +3520,22 @@ _CONSTEXPR20 pair<_RanIt, _RanIt> _Partition_by_median_guess_unchecked(_RanIt _F
 template <class _RanIt, class _Pr>
 _CONSTEXPR20_ICE void _Sort_unchecked(_RanIt _First, _RanIt _Last, _Iter_diff_t<_RanIt> _Ideal, _Pr _Pred) {
     // order [_First, _Last), using _Pred
-    _Iter_diff_t<_RanIt> _Count = 0;
-    while (_ISORT_MAX < (_Count = _Last - _First) && 0 < _Ideal) { // divide and conquer by quicksort
+    for (;;) {
+        if (_Last - _First < _ISORT_MAX) { // small

```

Comment:
There are two behavior changes here:
1. We previously didn't call `_Insertion_sort_unchecked` when `_Count < 2`. We'll now perform another iterator compare or two, and maybe an increment. This is probably a wash, and no one will notice.
2. We previously called `_Insertion_sort_unchecked` when `_Last - _First == _ISORT_MAX`. This might be noticeable for some use cases, and is super easy to fix by changing `<` to `<=` here.

Reply:
1. We perform 1 extra iterator compare in `_Insertion_sort_unchecked` in exchange for performing one less integer compare here.
2. Agreed. I failed at inverting the condition I guess.

Reply:
Hmmm I missed that. Now I think it's worth changing :)